xl_auth/commands.py

```diff
@@ -172,3 +172,254 @@ def urls(url, order):
+                voyager_sigels_and_collections[voyager_sigel].add(voyager_location)
+            else:
+                voyager_sigels_and_collections[voyager_sigel] = {voyager_location}
+        print('voyager_main_sigels:', len(voyager_main_sigels), '/',
+              'voyager_location_sigels:', len(voyager_location_sigels))
+        print('(voyager_main_sigels | voyager_location_sigels):',
+              len(voyager_main_sigels | voyager_location_sigels))
+
+        return {
+            'sigel_to_collections': voyager_sigels_and_collections,
+            'sigels': voyager_main_sigels,
+            'collections': voyager_location_sigels
+        }
+
+    def _get_bibdb_data():
+        raw_bibdb_sigels_and_cataloging_admins = requests.get(
+            'https://libris.kb.se/libinfo/library_konreg.jsp').content.decode().splitlines()
+
+        registering_bibdb_sigels, bibdb_cataloging_admins = set(), set()
+        bibdb_sigels_and_cataloging_admins = dict()
+        bibdb_cataloging_admins_and_sigels = dict()
+        bibdb_cataloging_admin_emails_and_names = dict()
+        for bibdb_row in raw_bibdb_sigels_and_cataloging_admins:
+            bibdb_sigel, cataloging_admin_name, cataloging_admin_email = bibdb_row.split(',')
+            cataloging_admin_email = cataloging_admin_email.lower()
+            bibdb_cataloging_admin_emails_and_names[cataloging_admin_email] = cataloging_admin_name
+            assert bibdb_sigel != ''
+            registering_bibdb_sigels.add(bibdb_sigel)
+            if not cataloging_admin_email:
+                continue
+
+            bibdb_cataloging_admins.add(cataloging_admin_email)
+            bibdb_sigels_and_cataloging_admins[bibdb_sigel] = cataloging_admin_email
+
+            if cataloging_admin_email in bibdb_cataloging_admins_and_sigels:
+                bibdb_cataloging_admins_and_sigels[cataloging_admin_email].add(bibdb_sigel)
+            else:
+                bibdb_cataloging_admins_and_sigels[cataloging_admin_email] = {bibdb_sigel}
+
+        print('registering_bibdb_sigels:', len(registering_bibdb_sigels), '/',
+              'bibdb_cataloging_admins:', len(bibdb_cataloging_admins))
+
+        return {
+            'sigel_to_cataloging_admins': bibdb_sigels_and_cataloging_admins,
+            'cataloging_admin_to_sigels': bibdb_cataloging_admins_and_sigels,
+            'cataloging_admin_emails_to_names': bibdb_cataloging_admin_emails_and_names,
+            'sigels': registering_bibdb_sigels,
+            'cataloging_admins': bibdb_cataloging_admins,
+        }
+
+    def _get_bibdb_sigels_not_in_voyager(bibdb_sigels, voyager_sigels):
+        unknown_sigels = set()
+        for bibdb_sigel in bibdb_sigels:
+            if bibdb_sigel not in voyager_sigels:
+                unknown_sigels.add(bibdb_sigel)
+        return unknown_sigels
+
+    def _generate_xl_auth_cataloging_admins_and_collections(bibdb_cataloging_admin_to_sigels,
```

Comment:
There are a bunch of `print()` and `click.echo()` statements here that only print a variable, making the output difficult to interpret. Can you add some more info there?

This (and maybe some other functions too) can benefit from some comments, outlining the different steps.

Reply:
Done!